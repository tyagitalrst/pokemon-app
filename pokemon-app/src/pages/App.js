import React from "react";
import { Route, BrowserRouter as Router, Switch } from "react-router-dom";
import PokemonProvider from "../providers/pokemon/provider";

import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import Theme from "../assets/theme";
import AppBar from "../components/Shared/AppBar";
import PokemonList from "./List/PokemonList/PokemonList";
import PokemonDetail from "./Detail/PokemonDetail";
import PokemonCollectionList from "./List/PokemonCollectionList/PokemonCollectionList";


const App = () => {
  return (
    <Router>
      <PokemonProvider>
        <ThemeProvider theme={Theme}>
          <CssBaseline />
          <AppBar></AppBar>
          <Switch>
            <Route exact path="/" component={PokemonList} />
            <Route path="/detail/:name/:pokemonName?" component={PokemonDetail} />
            <Route path="/collections" component={PokemonCollectionList} />
          </Switch>
        </ThemeProvider>
      </PokemonProvider>
    </Router>
  );
};

export default App;
