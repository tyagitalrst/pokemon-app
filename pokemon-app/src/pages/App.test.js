import React from "react";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";
import App from "./App";

afterEach(cleanup);

describe("App.js", () => {
  test("Snapshot Test App", () => {
    const component = renderer.create(
      <MockedProvider>
        <App />
      </MockedProvider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
