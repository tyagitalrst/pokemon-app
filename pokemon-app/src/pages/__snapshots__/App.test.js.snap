// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js Snapshot Test App 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
      >
        Pokédex
      </h6>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Pokemons
        </span>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        href="/collections"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          My Pokemons
        </span>
      </a>
    </div>
  </header>,
  <div
    className="pokemon-list"
  >
    <div>
      <div
        className="MuiPaper-root makeStyles-pokemonHeader-7 MuiPaper-elevation3 MuiPaper-rounded"
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
          >
            <div
              className="makeStyles-headerImage-6"
            >
              <img
                alt="pokeball"
                src="pokeball.png"
              />
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-headerInfo-8 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
          >
            <div>
              <h2
                className="MuiTypography-root MuiTypography-h2 MuiTypography-colorPrimary"
              >
                Pokémon List
              </h2>
            </div>
            <div>
              <h4
                className="MuiTypography-root MuiTypography-h4 MuiTypography-colorSecondary"
              >
                Total Owned Pokemons: 0 Pokemon
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root makeStyles-pagination-5 MuiGrid-container MuiGrid-justify-content-xs-space-evenly"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-colorSecondary"
        >
          Show 1-0 from 0
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <nav
          aria-label="pagination navigation"
          className="MuiPagination-root"
        >
          <ul
            className="MuiPagination-ul"
          >
            <li>
              <button
                aria-label="Go to previous page"
                className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textSecondary Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiPaginationItem-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-current="true"
                aria-label="page 1"
                className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textSecondary Mui-selected"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                1
              </button>
            </li>
            <li>
              <button
                aria-label="Go to next page"
                className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textSecondary Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiPaginationItem-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </nav>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-space-between"
    />
  </div>,
]
`;
